USING_DOCKER.md

```diff
@@ -0,0 +1,93 @@
+# Using the Dockerfile
```

Comment:
I think we should put this in `DEVELOPING.md`, it fits well and will be easy to find there. WDYT?